autoload/intero/process.vim

```diff
@@ -66,7 +66,13 @@ function! intero#process#initialize() abort
         if (!exists('g:intero_stack_yaml'))
             " Change dir temporarily and see if stack can find a config
             silent! lcd %:p:h
```

Comment:
The current behavior makes a lot of sense. Suppose I am in `~` and I open `~/Projects/some-dir/Foo.hs`. If there's a `~/Projects/some-dir/stack.yaml`, then I don't want it to pick up the global Stack resolver, I want it to pick up the most appropriate `stack.yaml` based on the file.

I think we can actually do better here: vim allows us to access environment variables, so we can just check `$STACK_YAML` for being a relative or absolute path. If it's absolute, then we just use it as-is. If it's relative, then we try current directory to get the absolute path.